From 53944d3b20d7b4e2b3a6c3c19942c970615c45df Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Tue, 28 Aug 2018 18:22:25 -0600
Subject: [PATCH 31/44] soong_config: Add add_json_str_omitempty function

* This function wraps add_json_str so that it does not add the soong
  variable key:value pair to json structure if the second argument
  (value) is empty/unset, preventing the need for more complex
  conditional logic in the middle the code that builds the struct
* This concept allows us to make use of the omitempty field option
  for purposes like single variables controlling ifdef blocks as well
  as carrying the desired value

Change-Id: I99c8c162069c2aa8ff3d0bab2636b01181e74a9d
---
 common/json.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/json.mk b/common/json.mk
index ba8ffa7..e0aac9a 100644
--- a/common/json.mk
+++ b/common/json.mk
@@ -20,6 +20,7 @@ json_decrease_indent =$= $(eval _json_indent := $$(subst _,$$(space),$$(patsubst
 # 2: Value
 add_json_val =$= $(eval _json_contents := $$(_json_contents)$$(_json_indent)"$$(strip $$(1))": $$(strip $$(2))$$(comma)$$(newline))
 add_json_str =$= $(call add_json_val,$(1),"$(strip $(2))")
+add_json_str_omitempty =$= $(if $(strip $(2)),$(call add_json_str, $(1), $(2)))
 add_json_list =$= $(call add_json_val,$(1),$(call json_list,$(patsubst %,%,$(2))))
 add_json_csv =$= $(call add_json_val,$(1),$(call csv_to_json_list,$(strip $(2))))
 add_json_bool =$= $(call add_json_val,$(1),$(if $(strip $(2)),true,false))
-- 
2.7.4

